src/main/java/mate/fruitshop/service/strategy/TransactionHandlerStrategy.java

```diff
@@ -0,0 +1,17 @@
+package mate.fruitshop.service.strategy;
+
+import java.util.HashMap;
+import java.util.Map;
+import mate.fruitshop.model.FruitTransaction;
+
+public class TransactionHandlerStrategy {
+    public static final Map<FruitTransaction.Operation, FruitTransactionHandler>
+            TRANSACTIONS_MAP = new HashMap<>();
+
+    static {
+        TRANSACTIONS_MAP.put(FruitTransaction.Operation.BALANCE, new FruitBalanceHandler());
+        TRANSACTIONS_MAP.put(FruitTransaction.Operation.SUPPLY, new FruitSupplyHandler());
+        TRANSACTIONS_MAP.put(FruitTransaction.Operation.PURCHASE, new FruitPurchaseHandler());
+        TRANSACTIONS_MAP.put(FruitTransaction.Operation.RETURN, new FruitReturnHandler());
+    }
```

Comment:
you used constant case for this map, but it's not a constant, because static final isn't enough, it should be deeply immutable
You can use Map.of, it will create immutable map, and you won't need static init block